air-quality-ui/system_tests/ui/summary_page.spec.ts

```diff
@@ -1,75 +1,807 @@
+        pm2_5: { mean: { aqi_level: 5, value: 52.75 } },
+        pm10: { mean: { aqi_level: 4, value: 58.25755291558235 } },
+        so2: { mean: { aqi_level: 1, value: 8.497931484924965 } },
+      }),
+      createMeasurementSummaryAPIResponseData({
+        measurement_base_time: '2024-06-19T12:00:00Z',
+        location_name: 'Zurich',
+        overall_aqi_level: { mean: 2 },
+        no2: { mean: { aqi_level: 1, value: 1.2220194497781245 } },
+        o3: { mean: { aqi_level: 3, value: 110.29793453644987 } },
+        pm2_5: { mean: { aqi_level: 2, value: 15.764618078867594 } },
+        pm10: { mean: { aqi_level: 2, value: 31.71802172436572 } },
+        so2: { mean: { aqi_level: 1, value: 1.3459434727665889 } },
+      }),
+    ]
+    await vairifySummaryPage.setupPageWithMockData(
+      mockedForecastResponse,
+      mockedMeasurementSummaryResponse,
+    )
+  })
+  test.describe('Table Structure', () => {
+    test('Verify that Headers exist and Innertext matches', async ({
+      vairifySummaryPage,
+    }) => {
+      await vairifySummaryPage.getColumnHeaderAndText('AQI Level', 'AQI Level')
+      await vairifySummaryPage.getColumnHeaderAndText(
+        'PM 2.5 (µg/m³)',
+        'PM 2.5 (µg/m³)',
+      )
+      await vairifySummaryPage.getColumnHeaderAndText(
+        'PM 10 (µg/m³)',
+        'PM 10 (µg/m³)',
+      )
+      await vairifySummaryPage.scrollToRightmostPosition()
+      await vairifySummaryPage.page.waitForTimeout(1000)
+
+      await vairifySummaryPage.getColumnHeaderAndText(
+        'Nitrogen Dioxide (µg/m³)',
+        'Nitrogen Dioxide (µg/m³)',
+      )
+      await vairifySummaryPage.getColumnHeaderAndText(
+        'Ozone (µg/m³)',
+        'Ozone (µg/m³)',
+      )
+      await vairifySummaryPage.getColumnHeaderAndText(
+        'Sulphur Dioxide (µg/m³)',
+        'Sulphur Dioxide (µg/m³)',
+      )
+    })
+
+    test('Verify numbers in cells have no more than 1 decimal place ', async ({
+      vairifySummaryPage,
+    }) => {
+      await vairifySummaryPage.checkCellNumberFormat()
+    })
+  })
+
+  test.describe('Table Data Validation', () => {
+    test('Verify that a city with no in-situ data still show on grid', async ({
+      vairifySummaryPage,
+    }) => {
+      const count = await vairifySummaryPage.textCellSearch('Kyiv')
+      expect(count).toEqual(1)
+    })
+
+    test('Check data is displayed correctly on grid', async ({
+      vairifySummaryPage,
+    }) => {
+      // first 6 values for Kampala, Abu Dhabi, Zurich and Kyiv respectively
+      const expectedData = [
+        ['2', '6', '-4', '16.1', '76', '19 Jun 09:00'],
+        ['4', '5', '-1', '30.3', '52.8', '19 Jun 12:00'],
+        ['2', '2', '0', '17.2', '15.8', '19 Jun 12:00'],
+        ['2', '-', '-', '7', '-', '24 Jun 09:00'],
+      ]
+
+      await vairifySummaryPage.assertGridValues(expectedData)
+    })
+
+    test('Verify that Diff displays the delta between forcast and measured', async ({
```

Comment:
This is also covered above?

Reply:
This was also already in here, I agree, and am removing